tests/jballerina-unit-test/src/test/resources/test-src/types/table/record-type-table-key.bal

```diff
@@ -201,6 +201,85 @@ function testVariableNameFieldAsKeyField() {
     assertEquality(expected, tb.toString());
 }
 
+function testDefaultValueFieldAsKeyField() {
+    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
+    table<record {readonly int id = 1; string name;}> key (id) tb = table [
+        {id, name: "Jo"},
+        {id: 2, name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+function testDefaultValueFieldAsKeyField2() {
+    string expected = "[{\"id\":1,\"name\":\"Amy\"},{\"id\":1,\"name\":\"Jo\"}]";
+    table<record {readonly int id = 1; readonly string name = "Amy";}> key (id, name) tb = table [
+        {},
+        {name:"Jo"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+function testDefaultValueFieldAsKeyField3() {
+    string expected = "[{\"id\":1,\"name\":\"Amy\",\"key\":2},{\"id\":1,\"name\":\"Jo\",\"key\":2}]";
+    table<record {readonly int id = 1; readonly string name = "Amy"; readonly int key = 2;}> key (id, name) tb = table [
+        {},
+        {name:"Jo"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+type CustomRecord record {
+    readonly int id = 1;
+    string name;
+};
+
+function testDefaultValueFieldAsKeyField4() {
+    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
+    table<CustomRecord> key (id) tb = table [
+        {id, name: "Jo"},
+        {id: 2, name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+type CustomRecord2 record {
+    readonly int id = 1;
+    readonly string status = "status - 1";
+    string name;
+};
+
+const string status = "status - 1";
+
+function testDefaultValueFieldAsKeyField5() {
+    string expected =
+        "[{\"id\":1,\"status\":\"status - 1\",\"name\":\"Jo\"},{\"id\":2,\"status\":\"status - 2\",\"name\":\"Amy\"}]";
+    table<CustomRecord2> key (id, status) tb = table [
+        {id, status, name: "Jo"},
+        {id: 2, status: "status - 2", name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+function testDefaultValueFieldAsKeyField6() {
+    string expected =
+        "[{\"id\":1,\"status\":\"status - 1\",\"name\":\"Jo\"},{\"id\":1,\"status\":\"status - 2\",\"name\":\"Amy\"},{\"id\":3,\"status\":\"status - 1\",\"name\":\"Amy\"}]";
+    table<CustomRecord2> key (id, status) tb = table [
+        {name: "Jo"},
+        {status: "status - 2", name: "Amy"},
+        {id: 3, name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+function runTestCasesWithDefaultValueFieldAsKeyField() {
+    testDefaultValueFieldAsKeyField();
+    testDefaultValueFieldAsKeyField2();
+    testDefaultValueFieldAsKeyField3();
+    testDefaultValueFieldAsKeyField4();
+    testDefaultValueFieldAsKeyField5();
+    testDefaultValueFieldAsKeyField6();
+}
```

Comment:
We try to use a data provider on the Java side with multiple tests like these so that the remaining tests can run even when an initial test fails.

Reply:
Updated. Please check